x/gamm/keeper/pool_service_test.go

```diff
@@ -222,6 +241,147 @@ func (suite *KeeperTestSuite) TestCreateBalancerPool() {
+				err = gammKeeper.InitializePool(suite.Ctx, test.pool(), sender)
+
+				if test.expectPass {
+					suite.Require().NoError(err, "test: %v", test.name)
+
+					// check to make sure new pool exists and has minted the correct number of pool shares
+					pool, err := gammKeeper.GetPoolAndPoke(suite.Ctx, defaultPoolId)
+					suite.Require().NoError(err, "test: %v", test.name)
+					suite.Require().Equal(types.InitPoolSharesSupply.String(), pool.GetTotalShares().String(),
+						fmt.Sprintf("share token should be minted as %s initially", types.InitPoolSharesSupply.String()),
+					)
+
+					// get expected tokens in new pool and corresponding pool shares
+					expectedPoolTokens := sdk.Coins{}
+					for _, asset := range test.pool().GetTotalPoolLiquidity(suite.Ctx) {
+						expectedPoolTokens = expectedPoolTokens.Add(asset)
+					}
+					expectedPoolShares := sdk.NewCoin(types.GetPoolShareDenom(pool.GetId()), types.InitPoolSharesSupply)
+
+					// make sure expected pool tokens and expected pool shares matches the actual tokens and shares in the pool
+					suite.Require().Equal(expectedPoolTokens.String(), pool.GetTotalPoolLiquidity(suite.Ctx).String())
+					suite.Require().Equal(expectedPoolShares.Amount.String(), pool.GetTotalShares().String())
+
+					// check pool metadata
+					poolShareBaseDenom := types.GetPoolShareDenom(pool.GetId())
+					poolShareDisplayDenom := fmt.Sprintf("GAMM-%d", pool.GetId())
+					metadata, found := bankKeeper.GetDenomMetaData(suite.Ctx, poolShareBaseDenom)
+					suite.Require().Equal(found, true, fmt.Sprintf("Pool share denom %s is not set", poolShareDisplayDenom))
+					suite.Require().Equal(metadata.Base, poolShareBaseDenom, fmt.Sprintf("Pool share base denom %s is not correctly set", poolShareBaseDenom))
+					suite.Require().Equal(metadata.Display, poolShareDisplayDenom, fmt.Sprintf("Pool share display denom %s is not correctly set", poolShareDisplayDenom))
+					suite.Require().Equal(metadata.DenomUnits[0].Denom, poolShareBaseDenom)
+					suite.Require().Equal(metadata.DenomUnits[0].Exponent, uint32(0x0))
+					suite.Require().Equal(metadata.DenomUnits[0].Aliases, []string{
+						"attopoolshare",
+					})
+					suite.Require().Equal(metadata.DenomUnits[1].Denom, poolShareDisplayDenom)
+					suite.Require().Equal(metadata.DenomUnits[1].Exponent, uint32(types.OneShareExponent))
+					suite.Require().Equal(metadata.DenomUnits[1].Aliases, []string(nil))
+
+					// check AfterPoolCreated hook
+					for _, lockableDuration := range poolIncentivesKeeper.GetLockableDurations(suite.Ctx) {
+						gaugeId, err := poolIncentivesKeeper.GetPoolGaugeId(suite.Ctx, defaultPoolId, lockableDuration)
+						suite.Require().NoError(err, "test: %v", test.name)
+
+						poolIdFromPoolIncentives, err := poolIncentivesKeeper.GetPoolIdFromGaugeId(suite.Ctx, gaugeId, lockableDuration)
+						suite.Require().NoError(err, "test: %v", test.name)
+						suite.Require().Equal(poolIdFromPoolIncentives, defaultPoolId)
+					}
+
```

Comment:
Please also add a check that `sender` had shares minted to them by checking their address's bank balance